src/cache.rs

```diff
@@ -289,16 +289,25 @@ impl Cache {
         let patch_path = Self::find_patch(&patch_filename, custom_pages_dir);
 
         // Try to find a platform specific path next, append custom patch to it.
-        let platform_dir = self.get_platform_dir();
-        if let Some(page) =
-            Self::find_page_for_platform(&page_filename, &cache_dir, platform_dir, &lang_dirs)
-        {
-            return Some(PageLookupResult::with_page(page).with_optional_patch(patch_path));
+        let current_platform = self.get_platform_dir();
+
+        let mut platforms = PlatformType::get_platforms();
+        platforms.push("common");
+
+        // remove the current platform from the vector and re-add it to the end
+        let index = platforms.iter().position(|&p| p == current_platform).unwrap();
+        platforms.remove(index);
+        platforms.push(current_platform);
+
+        // Cycle through all platforms in reverse order (current_platform - common - all other platforms)
+        for platform in platforms.iter().rev() {
+            if let Some(page) =
+                Self::find_page_for_platform(&page_filename, &cache_dir, platform, &lang_dirs)
+                {
+                    return Some(PageLookupResult::with_page(page).with_optional_patch(patch_path));
+                }
         }
-
-        // Did not find platform specific results, fall back to "common"
-        Self::find_page_for_platform(&page_filename, &cache_dir, "common", &lang_dirs)
-            .map(|page| PageLookupResult::with_page(page).with_optional_patch(patch_path))
+        return None;
```

Comment:
`clippy` tells us that this `return` is unneeded. In general, when handling `Some`, `None`, iterators, etc. manually, there is a good chance that there is already some helper function from the standard library that does exactly what we want. In this case, it is `find_map`. Using that, we can express the above as:
```rust
directories_to_search
    .into_iter()
    .find_map(|platform_dir| {
        Self::find_page_for_platform(&page_filename, &cache_dir, platform_dir, &lang_dirs)
    })
    .map(|page| PageLookupResult::with_page(page).with_optional_patch(patch_path))
```

This is nice! But there is one more flaw in there. We are using `into_iter` on our freshly created vector. Maybe we can spare the allocation? Of course, the creation of our `directories_to_search` is basically just building an iterator and collecting it, so we can just skip the "collecting" step. Finally, let's use this opportunity to also give names to the iterators themselves and remove the comments - now, everything is clear from the code:

```rust
let main_directories = [current_platform_dir, "common"];
let secondary_directories = PlatformType::all()
    .iter()
    .copied()
    .filter(|&p| p != self.platform)
    .map(Self::directory_name_for);
let mut directories_to_search = main_directories
    .iter()
    .copied()
    .chain(secondary_directories);
```
Now, we can remove the `into_iter` below and arrive at what I believe to be the optimal solution.